modules/accumulo/src/main/java/org/apache/fluo/recipes/accumulo/export/AccumuloWriter.java

```diff
@@ -0,0 +1,310 @@
+          exports.add(exportQueue.take());
+          exportQueue.drainTo(exports);
+
+          for (Mutations ml : exports) {
+            bw.addMutations(ml.mutations);
+          }
+
+          bw.flush();
+
+          // notify all threads waiting after flushing
+          for (Mutations ml : exports) {
+            ml.cdl.countDown();
+          }
+
+        } catch (InterruptedException | MutationsRejectedException e) {
+          throw new RuntimeException(e);
+        }
+      }
+    }
+
+  }
+
+  private static LinkedBlockingQueue<Mutations> exportQueue = null;
+
+  private AccumuloWriter(String instanceName, String zookeepers, String user, String password,
+      String table) {
+
+    // TODO: fix this write to static and remove findbugs max rank override in pom.xml
+    exportQueue = new LinkedBlockingQueue<>(10000);
+
+    try {
+      Thread queueProcessingTask =
+          new Thread(new ExportTask(instanceName, zookeepers, user, password, table));
+      queueProcessingTask.setDaemon(true);
+      queueProcessingTask.start();
+    } catch (Exception e) {
+      throw new IllegalStateException(e);
+    }
+  }
+
+  private static Map<String, AccumuloWriter> exporters = new HashMap<>();
+
+  public static AccumuloWriter getInstance(FluoConfiguration config, String id) {
+    return getInstance(config.getAppConfiguration(), id);
+  }
+
+  public static AccumuloWriter getInstance(SimpleConfiguration conf, String id) {
+    return getInstance(getConfig(conf, id));
+  }
+
+  public static AccumuloWriter getInstance(Config ti) {
+    return getInstance(ti.instanceName, ti.zookeepers, ti.user, ti.password, ti.table);
+  }
+
+  public static synchronized AccumuloWriter getInstance(String instanceName, String zookeepers,
+      String user, String password, String table) {
+
+    String key =
+        instanceName + ":" + zookeepers + ":" + user + ":" + password.hashCode() + ":" + table;
+
+    AccumuloWriter ret = exporters.get(key);
+
+    if (ret == null) {
+      ret = new AccumuloWriter(instanceName, zookeepers, user, password, table);
+      exporters.put(key, ret);
+    }
+
+    return ret;
+  }
+
+  public void write(Collection<Mutation> mutations) {
+    Mutations work = new Mutations(mutations);
+    exportQueue.add(work);
+    try {
+      work.cdl.await();
+    } catch (InterruptedException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  /**
+   * Returns Replication Filter
+   */
+  public static Predicate<LogEntry> getReplicationFilter() {
+    return le -> le.getOp().equals(LogEntry.Operation.DELETE)
+        || le.getOp().equals(LogEntry.Operation.SET);
+  }
+
+  /**
+   * Generates Accumulo mutations from a Transaction log. Used to Replicate Fluo table to Accumulo.
+   *
+   * @param txLog Transaction log
+   * @param seq Export sequence number
+   * @return Collection of mutations
+   */
+  public static Collection<Mutation> generateMutations(long seq, TxLog txLog) {
```

Comment:
Yes. I agree.  The `TxLog` code is in fluo-recipes-core but this is here to avoid adding accumulo as dependency to core.  I will make a `AccumuloReplication` class fluo-recipes-accumulo and add this and the filter method above.  Does that work?